src/win/getaddrinfo.c

```diff
@@ -126,34 +115,44 @@ static void uv__getaddrinfo_done(struct uv__work* w, int status) {
   }
 
   if (req->retcode == 0) {
+    char *alloc_ptr = NULL;
+    size_t cur_off = 0;
+    size_t addrinfo_len;
     /* Convert addrinfoW to addrinfo. First calculate required length. */
-    addrinfow_ptr = req->addrinfow;
+    struct addrinfoW *addrinfow_ptr = req->addrinfow;
     while (addrinfow_ptr != NULL) {
-      addrinfo_len += addrinfo_struct_len +
-          ALIGNED_SIZE(addrinfow_ptr->ai_addrlen);
+      align_offset(&cur_off, sizeof(void*));
+      cur_off += sizeof(struct addrinfo);
+      /* TODO: This alignment could be smaller, if we could
+	           portably get the alignment for sockaddr. */
+      align_offset(&cur_off, sizeof(void*));
+      cur_off += addrinfow_ptr->ai_addrlen;
       if (addrinfow_ptr->ai_canonname != NULL) {
-        name_len = uv_utf16_length_as_wtf8(addrinfow_ptr->ai_canonname, -1);
+        ssize_t name_len =
+            uv_utf16_length_as_wtf8(addrinfow_ptr->ai_canonname, -1);
         if (name_len < 0) {
           req->retcode = name_len;
           goto complete;
         }
-        addrinfo_len += ALIGNED_SIZE(name_len + 1);
+        cur_off += name_len + 1;
       }
       addrinfow_ptr = addrinfow_ptr->ai_next;
     }
 
     /* allocate memory for addrinfo results */
-    alloc_ptr = (char*)uv__malloc(addrinfo_len);
+    addrinfo_len = cur_off;
+    alloc_ptr = (char *)uv__malloc(addrinfo_len);
 
     /* do conversions */
     if (alloc_ptr != NULL) {
-      cur_ptr = alloc_ptr;
+      struct addrinfo *addrinfo_ptr = (struct addrinfo *)alloc_ptr;
```

Comment:
style nit: (but also the declaration needs to be at the top, separate from the declaration)
```suggestion
      struct addrinfo* addrinfo_ptr = (struct addrinfo*) alloc_ptr;
```

Reply:
I can't parse that sentence. You mean instead of a declaration with an initializer, do an uninitialized declaration, and then a separate assignment later on?

And at the top of what? Block, Function, File?

FWIW this change fits with existing code in the file, which declares variables with initialization in block scope.
But also, these style issues could be enforced with tooling.

Reply:
Yes, someday we should make a clang-format file so that these issues are easier to deal with. Hasn't been top of anyone's list I suppose.

It should go at the top of the function, though sometimes top of the scope gets used instead.

And yes, it is C89-standard to write declarations with initialization, then separately assign it later. It is mostly a style thing, but the style avoids annoying refactoring issues with the first assignment being distinct when variables get reused (eg. for repeated status codes) and it avoids accidentally shadowing variables from outer scopes.